tuftool/src/create.rs

```diff
@@ -77,202 +63,49 @@ pub(crate) struct CreateArgs {
-                    }
-                },
-                _extra: HashMap::new(),
-            },
-            self.args.timestamp_version,
-            "timestamp.json",
-        )?;
+        let targets = self.build_targets()?;
+        let mut editor =
+            RepositoryEditor::new(&self.root).context(error::EditorCreate { path: &self.root })?;
+
+        editor
+            .add_targets(targets)
+            .context(error::TargetsFromDir { dir: &self.indir })?
+            .targets_version(self.targets_version)
+            .targets_expires(self.targets_expires)
+            .snapshot_version(self.snapshot_version)
+            .snapshot_expires(self.snapshot_expires)
+            .timestamp_version(self.timestamp_version)
+            .timestamp_expires(self.timestamp_expires);
+
+        let signed_repo = editor.sign(&self.keys).context(error::SignRepo)?;
+
+        let metadata_dir = &self.outdir.join("metadata");
+        let targets_dir = &self.outdir.join("targets");
+        signed_repo.write(metadata_dir).context(error::WriteRepo)?;
+        signed_repo
+            .link_targets(&self.indir, targets_dir)
+            .context(error::LinkTarget {
+                indir: &self.indir,
+                outdir: targets_dir,
+            })?;
 
         Ok(())
     }
 
-    // =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
-
-    fn build_targets(&self) -> Result<HashMap<String, Target>> {
-        WalkDir::new(&self.args.indir)
-            .follow_links(self.args.follow)
-            .into_iter()
-            .par_bridge()
-            .filter_map(|entry| match entry {
-                Ok(entry) => {
-                    if entry.file_type().is_file() {
-                        Some(self.process_target(entry.path()))
-                    } else {
-                        None
-                    }
-                }
-                Err(err) => Some(Err(err).context(error::WalkDir)),
-            })
-            .collect()
-    }
-
-    fn process_target(&self, path: &Path) -> Result<(String, Target)> {
-        let target_name = path.strip_prefix(&self.args.indir).context(error::Prefix {
-            path,
-            base: &self.args.indir,
-        })?;
-        let target_name = target_name
-            .to_str()
-            .context(error::PathUtf8 { path: target_name })?
-            .to_owned();
-
-        let mut file = File::open(path).context(error::FileOpen { path })?;
-        let mut digest = Context::new(&SHA256);
-        let mut buf = [0; 8 * 1024];
-        let mut length = 0;
-        loop {
-            match file.read(&mut buf).context(error::FileRead { path })? {
-                0 => break,
-                n => {
-                    digest.update(&buf[..n]);
-                    length += n as u64;
-                }
+    // Build a list of files from the given directory
+    fn build_targets(&self) -> Result<Vec<PathBuf>> {
```

Comment:
```suggestion
    fn find_targets(&self) -> Result<Vec<PathBuf>> {
```
Not really building anymore...